File: src/c/merge_sorted/merge_sorted.c
#include <stddef.h>

void merge_sorted(int* a1, size_t len1, int* a2, size_t len2, int* a)
{
    size_t i, k, l;
    i = k = l = 0;

    while(i<len1 || k<len2)
    {
        if(i<len1)
        {
            if((k<len2 && a1[i]<=a2[k]) || k>=len2)
            {
                a[l]=a1[i];
                l++;
                i++;
            }
        }
        if(k<len2)
        {
            if((i<len1 && a2[k]<=a1[i]) || i>=len1)
            {
                a[l]=a2[k];
                l++;
                k++;
            }
        }
    }
}

File: src/js/merge_sorted.js
function merge_sorted(array1, array2)
{
    let len1 = array1.length;
    let len2 = array2.length;
    let result = new Array(len1 + len2);

    let i, k, l;
    i = k = l = 0;

    while(i<len1 || k < len2)
    {
        if(i<len1)
        {
            if(k>=len2 ||  array1[i]<=array2[k])
            {
                result[l] = array1[i];
                i++;
                l++;
            }
        }
        if(k<len2)
        {
            if(i>=len1 ||  array2[k]<=array1[i])
            {
                result[l] = array2[k];
                k++;
                l++;
            }
        }
    }

    return result;
}